File: src/main/java/org/proofit/premiumcalc/calculation/FirePremium.java
package org.proofit.premiumcalc.calculation;

import java.math.BigDecimal;

public class FirePremium implements Premium {
    private static final BigDecimal DEFAULT_COEFFICIENT = new BigDecimal("0.014");
    private static final BigDecimal ALTERNATIVE_COEFFICIENT = new BigDecimal("0.024");
    private static final BigDecimal RISK_LIMIT = new BigDecimal("100.00");

    @Override
    public BigDecimal getCoefficient(BigDecimal sumInsured) {
        return shouldDefaultRiskBeUsedFor(sumInsured) ? DEFAULT_COEFFICIENT : ALTERNATIVE_COEFFICIENT;
    }

    private static boolean shouldDefaultRiskBeUsedFor(BigDecimal sumInsuredFire) {
        return RISK_LIMIT.compareTo(sumInsuredFire) > -1;
    }
}

File: src/main/kotlin/org/proofit/premiumcalc/calculation/FirePremium.kt
package org.proofit.premiumcalc.calculation

import java.math.BigDecimal

class FirePremium : Premium {
    private val DEFAULT_COEFFICIENT = BigDecimal("0.014")
    private val HIGH_RISK_COEFFICIENT = BigDecimal("0.024")
    private val RISK_LIMIT = BigDecimal("100.00")

    private fun isDefaultRisk(sumInsuredFire: BigDecimal): Boolean {
        return sumInsuredFire <= RISK_LIMIT
    }

    override fun getCoefficient(sumInsured: BigDecimal): BigDecimal {
        return if (isDefaultRisk(sumInsured)) DEFAULT_COEFFICIENT else HIGH_RISK_COEFFICIENT
    }
}
